core/src/util.jl

```diff
@@ -728,3 +722,23 @@ end
 function basin_areas(basin::Basin, state_idx::Int)
     return basin.level_to_area[state_idx].u
 end
+
+function fields_of_type(p::Parameters; type = AbstractDemandNode)
+    return (getfield(p, name) for name in propertynames(p) if getfield(p, name) isa type)
```

Comment:
```suggestion
    return (getfield(p, name) for name in fieldnames(p) if fieldtype(p, name) <: type)
```

Fields and properties don't neccesarily match, though they do for Parameters currently.